grails-app/controllers/org/pih/warehouse/api/InvoiceApiController.groovy

```diff
@@ -183,4 +183,12 @@ class InvoiceApiController {
         List<InvoiceItem> prepaymentItems = invoice.prepaymentItems
         render([data: prepaymentItems, totalCount: prepaymentItems.size()] as JSON)
     }
+
+    def validateInvoiceItem(InvoiceItem invoiceItem) {
+        if (!invoiceItem.validate(['quantity'])) {
```

Comment:
if we already touch the invoice item, shouldn't we try to validate it "deeply" (it means: not only the quantity, but whole object)?
If you agreed on validating the quantity only, then the method in my opinion should be called `validateInvoiceItemQuantity`, not `validateInvoiceItem`.

Reply:
```suggestion
        if (!invoiceItem.validate()) {
```

Reply:
If there will be a need for validating more than this one property we can add it, for now, it's unnecessary

Reply:
I'm unsure how I feel about this. @kchelstowski is right, but I don't want to change the scope of the method to the quantity. So I'd prefer keeping the scope at the invoice item, validating the whole object, and if necessary narrow the scope in the client. In other words, pull out just the quantity error message to display to the user.